packages/synapse-react-client/src/components/EntityUpload/EntityUploadModal.tsx

```diff
@@ -0,0 +1,69 @@
+import { Button } from '@mui/material'
+import React, { useState } from 'react'
+import { UploaderState } from '../../utils/hooks/useUploadFileEntity/useUploadFileEntities'
+import { DialogBase } from '../DialogBase'
+import { EntityUpload, EntityUploadHandle } from './EntityUpload'
+
+type EntityUploadModalProps = {
+  entityId: string
+  open: boolean
+  onClose: () => void
+}
+
+export const EntityUploadModal = React.forwardRef(function EntityUploadModal(
```

Comment:
Simple modal wrapper over the upload component. Uses the state to determine if the user is allowed to close it, i.e. you cannot close it while you have an upload in progress